ui-tests/src/test/java/org/openmrs/reference/page/AllergyPage.java

```diff
@@ -68,6 +78,68 @@ public void removeNoKnownAllergy() {
         driver.findElement(DELETE_NO_KNOWN).click();
     }
 
+    public  void clickOnAddNewAllergy(){
+        driver.findElement(ADD_NEW_ALLERGY).click();
+    }
+
+    public void chooseDrug(){
+        driver.findElement(CHOOSEDRUG).click();
+    }
+
+    public void chooseReaction(){
+        driver.findElement(CHOOSEREACTION).click();
+    }
+
+    public void selectSeverity(){
+        driver.findElement(SELECTSEVERITY).click();
+
+    public void allergyComment(){
+        setText(COMMENT,"I have malaria");
+    }
+
+    public void clickOnSaveButon(){
+        driver.findElement(SAVE).click();
+    }
+
+    public void clickOnCancelButon(){
+        driver.findElement(CANCEL).click();
+    }
+
+
+    public void clickOnEdditAllergy(){
+        driver.findElement(EDIT_ALLERGY).click();
+    }
+
+    public void selectAllergyType(int allergytype){
+        driver.findElement(ALLERGYTYPE).getTagName("label")[allergytype].click();
+    }
+
+    public String getSeclectedAllergernclass(int allergytype){
+        WebElement l = driver.findElement(AllERGEN).getTagName("ul")[allergytype];
```

Comment:
Its a good recommendation to use meaningful variable name in your code for easy code readability